lib/importproject.cpp

```diff
@@ -626,7 +626,7 @@ void ImportProject::importVcxproj(const std::string &filename, std::map<std::str
                     if (std::strcmp(e->Name(), "ClCompile") == 0) {
                         const char *include = e->Attribute("Include");
                         if (include && Path::acceptFile(include))
-                            compileList.push_back(include);
+                            compileList.emplace_back(include);
```

Comment:
I look forward to someday start using emplace instead of push when we can get some cleaner code.. however we currently do not use it for compatibility reasons..

however in this particular case it doesn't make the code much better as far as I see. Sometimes it means you don't have to explicitly create the type..

Reply:
clang-tidy suggests it in this (it somehow doesn't for a lot of other cases) and `emplace_back` is already used across the code even for cases where it doesn't save you the explicit type usage (see my other PR).

Reply:
See https://github.com/search?utf8=%E2%9C%93&q=emplace_back+user%3Adanmar+repo%3Adanmar%2Fcppcheck&type=Code&ref=advsearch&l=&l=

Reply:
ok.. for some containers we allow emplace/emplace_back.